src/splitter.ts

```diff
@@ -301,9 +313,33 @@ export const initializeSplitting = async (
       vehiclesInMessage,
       mainHeader,
       originMessageId.toString(),
-      sendCallback
+      sendCallback,
+      promises
     );
-    acknowledgeMessage();
+    Promise.all(promises)
+      .then((results) => {
+        if (
+          results.some(
+            (result: Pulsar.MessageId) => result.toString() === undefined
+          )
+        ) {
+          const messageIds = results.map((result) => result.toString());
+          logger.fatal(
+            { results, messageIds, originMessageId, pulsarTopic },
+            "Some messages were not sent"
+          );
+          process.exit(143);
+        } else {
+          acknowledgeMessage();
+        }
+      })
+      .catch(() => {
+        logger.fatal(
+          { originMessageId, pulsarTopic, promises },
+          "Some messages were not sent"
+        );
+        process.exit(143);
```

Comment:
Rather throw an exception and do not catch it yourself. That way the resources are closed before exiting.
```suggestion
        throw new Error("Some messages were not sent");
```